app/src/main/java/io/pslab/activity/MainActivity.java

```diff
@@ -344,6 +351,17 @@ private void setPSLabVersionIDs() {
         }
     }
 
+    public void showFirmwareDialog() {
+        AlertDialog.Builder builder = new AlertDialog.Builder(this);
+        builder.setTitle("Legacy Firmware Detected");
+        builder.setCancelable(false);
+        builder.setMessage("We have detected that your PSLab device is running legacy firmware. Please note that support for this firmware has ended. For the best experience and continued support, please update your device to the latest firmware version.");
```

Comment:
The message should be read from the strings.xml resource file instead of being hardcoded.

Reply:
Oh yes ! My bad. Just slipped out of my mind when I was working on this. I'll fix this and couple of other things tommorow. Thanks !!